### PR TITLE
feat: add executmany for python sdk

### PR DESCRIPTION
**Pull request by kitty-eu-org**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

No documentation found for executemany
tests for thanges have been  changed,because a related test already exists
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
[issue1389](https://github.com/4paradigm/OpenMLDB/issues/1389)

* **What is the new behavior (if this is a feature change)?**
no

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

### CODE REVIEW

File: python/openmldb/test/openmldb_client_test.py [hunk 1]
```diff
@@ -202,7 +204,7 @@ def check_exectute_many(self,connection,sql):
     try:
       connection.execute(sql,[{"col1":1005, "col2":"2020-12-29", "col3":"shandong", "col4":"jinan", "col5":6},
                               {"col1":1006, "col2":"2020-12-30", "col3":"fujian", "col4":"fuzhou", "col5":7}]);
-      self.assertTrue(False)
+      self.assertTrue(True)
```
Comment (vagetablechicken) [inline python/openmldb/test/openmldb_client_test.py]: If this part wroks fine, remove the try-except and this assert.

File: python/openmldb/dbapi/dbapi.py [hunk 2]
```diff
@@ -327,9 +343,103 @@ def execute(self, operation, parameters=()):
...
+                "Only {} is valid, params: {} are invalid, maybe not exists mark '?' in sql".format(operation,
+                                                                                                    parameters))
+            return self.execute(operation, parameters)
+        if isinstance(parameters, list) and len(parameters) == 0:
+            return self.execute(operation, parameters)
+
+        if insertRE.match(command):
+            question_mark_count = command.count("?")
+            if question_mark_count > 0:
+                # 因为getInsertBatchBuilder获得的对象没有GetSchema方法，所以使用getInsertBatchBuilder获得的对象
```
Comment (vagetablechicken) [inline python/openmldb/dbapi/dbapi.py]: plz use English

File: python/openmldb/dbapi/dbapi.py [hunk 3]
```diff
@@ -327,9 +343,103 @@ def execute(self, operation, parameters=()):
         else:
             raise DatabaseError("unsupport sql")
 
+    def __insert_rows(self, rows: List[Union[tuple, dict]], hold_idxs, schema, rows_builder, command):
+        for row in rows:
+            tmp_builder = rows_builder.NewRow()
```
Comment (vagetablechicken) [inline python/openmldb/dbapi/dbapi.py:322]: building one row is the same with https://github.com/4paradigm/OpenMLDB/blob/060120110a76a11806eb380fb5289a5e88ff7ab7/python/openmldb/dbapi/dbapi.py#L207-L220

reuse it. And you need to merge main, this file is changed.

**Review by vagetablechicken — CHANGES_REQUESTED** [review]
Need some fix

File: python/openmldb/test/openmldb_client_test.py [hunk 4]
```diff
@@ -199,12 +201,9 @@ def check_result(self, rs, expectRows, orderIdx = 0):
       i+=1
 
   def check_exectute_many(self,connection,sql):
-    try:
       connection.execute(sql,[{"col1":1005, "col2":"2020-12-29", "col3":"shandong", "col4":"jinan", "col5":6},
                               {"col1":1006, "col2":"2020-12-30", "col3":"fujian", "col4":"fuzhou", "col5":7}]);
-      self.assertTrue(False)
-    except Exception as e:
-      pass
+      self.assertTrue(True)
```
Comment (vagetablechicken) [inline python/openmldb/test/openmldb_client_test.py]: `self.assertTrue(True)` is needless

File: python/openmldb/dbapi/dbapi.py [hunk 5]
```diff
@@ -299,9 +275,93 @@ def execute(self, operation, parameters=()):
...
+                schema = builder.GetSchema()
+                hold_idxs = builder.GetHoleIdx()
+                for i in range(0, parameters_length, batch_number):
+                    rows = parameters[i: i + batch_number]
+                    ok, batch_builder = self.connection._sdk.getInsertBatchBuilder(self.db, command)
+                    if not ok:
+                        raise DatabaseError("get insert builder fail")
+                    self.__insert_rows(rows, hold_idxs, schema, batch_builder, command)
+            else:
+                ok, error = self.connection._sdk.executeInsert(self.db, command)
```
Comment (HuilinWu2) [inline python/openmldb/dbapi/dbapi.py]: we have `execute_sql(db, sql)` API that support all kinds of SQL without input params.

Comment (kitty-eu-org) [reply python/openmldb/dbapi/dbapi.py]: Do I need to make changes?

Comment (HuilinWu2) [reply python/openmldb/dbapi/dbapi.py]: I suppose so.

Comment (kitty-eu-org) [reply python/openmldb/dbapi/dbapi.py]: ok

**Review by vagetablechicken — APPROVED** [review]

**Review by HuilinWu2 — APPROVED** [review]

**Pull request merged**